python/neuroglancer/pipeline/storage.py

```diff
@@ -100,27 +65,44 @@ def extract_path(cls, layer_path):
         else:
             return cls.ExtractedPath(*match.groups())
 
-    def put_file(self, file_path, content, compress=True):
+    def put_file(self, file_path, content, compress=False):
         """ 
         Args:
             filename (string): it can contains folders
             content (string): binary data to save
         """
-        if compress:
-            content = self._compress(content)
+        return self.put_files([ (file_path, content) ], compress)
 
-        if self._n_threads:
-            # None is the non-existant callback
-            self._queue.put(('put_file', None, file_path, content, compress), block=True)
-        else:
-            self._interface(self._path).put_file(file_path, content, compress)
-        return self
+    def put_files(self, files, compress=False):
+        """
+        Put lots of files at once and get a nice progress bar. It'll also wait
+        for the upload to complete, just like get_files.
+
+        Required:
+            files: [ (filepath, content), .... ]
+        """
+        def base_uploadfn(path, content, interface):
+            interface.put_file(path, content, compress)
+
+        for path, content in files:
+            if compress:
+                content = self._compress(content)
+
+            uploadfn = partial(base_uploadfn, path, content)
 
+            if len(self._threads):
+                self.put(uploadfn)
+            else:
+                uploadfn(self._interface)
+
+        self.wait()
```

Comment:
I see, I guess I don't really understand the use of put_files, because to build the arguments you'll need some for loop, and you can use that for loop to directly call put_file instead.
Are we using put_files somewhere ?